docs/k8s/deployment-guide.mdx

```diff
@@ -0,0 +1,165 @@
+
+### Applying Labels and Annotations
+
+You may want to add specific labels or annotations to your resources, to help them be discovered and interact with other services like log scrapers or service meshes. You can set the `commonLabels` and `commonAnnotations` values for all resources created by the helm chart. Additionally, you can set annotations just on the pods themselves by setting the `podAnnotations` value.
+
+### Adding Extra Volumes and Environment Variables
+
+The helm chart also allows you to add extra volumes and environment variables to the operator, which is useful for mounting secrets or configmaps that contain credentials or other configuration values. This can be done by setting the `extraVolumes` and `extraEnv` values.
+
+### ngrok Region
+
+ngrok runs globally distributed tunnel servers around the world to enable fast, low latency traffic to your applications. See [ngrok's points of presence](/network-edge/#points-of-presence) for more information on ngrok's regions.
+
+Similar to the agent, if you do not explicitly pick a region via helm when installing the operator, the operator will attempt to pick the region with the least latency, which is usually the one geographically closest to your machine.
+
+See the [helm value `region`](https://github.com/ngrok/kubernetes-ingress-controller/blob/main/helm/ingress-controller/README.md#controller-parameters) to configure a specific region for the controller to use.
+
+## Metrics
+
+This operator exposes prometheus metrics on the `/metrics` endpoint. The metrics are exposed on the `:8080` port and can be scraped by prometheus or other services using typical means.
+
+This project is built using kube-builder, so out of the box it exposes the metrics listed [here](https://book.kubebuilder.io/reference/metrics-reference.html?highlight=metrics#default-exported-metrics-references)
+
+## White Label Agent Ingress
+
+If you're running the ngrok operator behind a corporate firewall, you may encounter connectivity issues due to the agent or controller not being able to communicate with the ngrok network edge. If you're facing such issues, you can run the ngrok cli command, which is documented [here](/guides/running-behind-firewalls) to diagnose the problem.
+
+However, if opening connectivity is not an option, you can set up a custom ingress domain on your dashboard. This way, you can configure the operator to use the custom domain instead of the default ngrok.app domain.
+
+To get started with this, go to your dashboard and create a custom ingress domain. Once created, you can configure the operator by using the following command:
+
+```bash
+helm install my-ingress-controller ngrok/kubernetes-ingress-controller \
+  --set serverAddr="ngrok.mydomain.com:443"
+```
+
```

Comment:
I would split the article here and try to capture advanced FAQs and gotchas in a separate article.  At this point the doc could stop and Review and link to the next advanced learning modules.

Reply:
I think this makes sense. The page was already ordered on a level of complexity, so I have split out the guide into a deployment guide and an "advanced deployments page" with stuff like whitelabeling ingress and multiple controllers.